File: frontend/src/app/services/cart.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

const CART_API = 'http://localhost:8080/api/v1/cart';

@Injectable({
  providedIn: 'root'
})
export class CartService {
  constructor(private http: HttpClient) {}

  getCart(): Observable<any[]> {
    return this.http.get<any[]>(CART_API);
  }

  addToCart(bookId: number, quantity: number): Observable<any> {
    let params = new HttpParams().set('bookId', bookId).set('quantity', quantity);
    return this.http.post(`${CART_API}/add`, {}, { params });
  }

  updateQuantity(id: number, quantity: number): Observable<any> {
    let params = new HttpParams().set('quantity', quantity);
    return this.http.put(`${CART_API}/update/${id}`, {}, { params });
  }

  removeItem(id: number): Observable<any> {
    return this.http.delete(`${CART_API}/remove/${id}`);
  }

  clearCart(): Observable<any> {
    return this.http.delete(`${CART_API}/clear`);
  }

  getTotal(): Observable<number> {
    return this.http.get<number>(`${CART_API}/total`);
  }
}

File: backend/src/main/java/com/booknest/cart/service/CartService.java
package com.booknest.cart.service;

import com.booknest.cart.entity.CartItem;
import com.booknest.user.entity.User;

import java.util.List;

public interface CartService {
    CartItem addToCart(User user, Long bookId, Integer quantity);
    CartItem updateQuantity(Long cartItemId, Integer quantity);
    void removeItem(Long cartItemId);
    List<CartItem> getCartItems(User user);
    void clearCart(User user);
    Double calculateTotal(User user);
}
